djvm/src/main/kotlin/net/corda/djvm/rewiring/KotlinMetadataVisitor.kt

```diff
@@ -0,0 +1,33 @@
+package net.corda.djvm.rewiring
+
+import org.objectweb.asm.AnnotationVisitor
+import org.objectweb.asm.commons.Remapper
+
+class KotlinMetadataVisitor(
+    api: Int,
+    av: AnnotationVisitor,
+    private val remapper: Remapper
+) : AnnotationVisitor(api, av) {
+    companion object {
+        val INTERNAL_NAME = "L([^.;\\[]++)+;".toRegex()
+    }
+
+    override fun visitArray(name: String): AnnotationVisitor? {
+        return super.visitArray(name)?.let {
+            if (name == "d2") KotlinMetadataTransformer(it) else it
```

Comment:
Maybe we could document the special role of `d2` and why this transformation is necessary.

Reply:
:+1: